js/jasmine/src/button.js

```diff
@@ -0,0 +1,11 @@
+function Button(buttonId, buttonAction) {
+  this.button       = document.getElementById(buttonId);
+  this.buttonAction = buttonAction;
+
+  this.initialize = function(inputId, outputId) {
+    var that = this;
+    this.button.addEventListener("click", function() {
+      that.buttonAction.act(inputId, outputId);
+    }, false);
+  };
+}
```

Comment:
There's a JavaScript way to create a constructor that could potentially combine `buttonId, buttonAction` args with `inputId, outputId`